File: RemootioAndroidAuto/automotive/src/main/java/com/example/remootioandroidws/RemootioScreen.java
package com.example.remootioandroidws;

import static androidx.car.app.model.Action.BACK;

import androidx.annotation.NonNull;
import androidx.car.app.CarContext;
import androidx.car.app.CarToast;
import androidx.car.app.Screen;
import androidx.car.app.model.ItemList;
import androidx.car.app.model.ListTemplate;
import androidx.car.app.model.Row;
import androidx.car.app.model.SectionedItemList;
import androidx.car.app.model.Template;

public class RemootioScreen extends Screen {
    public RemootioScreen(CarContext carContext) {
        super(carContext);
    }

    @NonNull
    @Override
    public Template onGetTemplate() {
        ListTemplate.Builder templateBuilder = new ListTemplate.Builder();

        ItemList radioList =
                new ItemList.Builder()
                        .addItem( new Row.Builder()
                                .setTitle("Test!")
                                .addText("click on any option to see toast notification")
                                .build())
                        .addItem(new Row.Builder().setTitle("Option 2").build())
                        .addItem(new Row.Builder().setTitle("Option 3").build())
                        .setOnSelectedListener(this::onSelected)
                        .build();
        templateBuilder.addSectionedList(
                SectionedItemList.create(radioList, "Sample list"));

        return templateBuilder.setTitle("List Selection Demo").setHeaderAction(BACK).build();
    }

    private void onSelected(int index) {
        CarToast.makeText(getCarContext(), "Changed selection to index: " + index, CarToast.LENGTH_LONG)
                .show();

    }
}

File: RemootioAndroidAuto/automotive/src/main/java/com/example/remootioandroidws/RemootioScreen.kt
package com.example.remootioandroidws

import androidx.car.app.CarContext
import androidx.car.app.CarToast
import androidx.car.app.Screen
import androidx.car.app.model.*

class RemootioScreen(carContext: CarContext?) : Screen(carContext!!) {
    override fun onGetTemplate(): Template {
        val templateBuilder = ListTemplate.Builder()
        val radioList = ItemList.Builder()
            .addItem(Row.Builder().setTitle("Garage Door").build())
            .addItem(Row.Builder().setTitle("Gate").build())
            .setOnSelectedListener { index: Int -> onSelected(index) }
            .build()
        templateBuilder.addSectionedList(
            SectionedItemList.create(radioList, "Remootio Devices")
        )
        return templateBuilder.setTitle("Remootio").build()
    }

    private fun onSelected(index: Int) {
        this.screenManager.push(RemootioDeviceScreen(carContext))
    }
}